dspace-spring-rest/src/test/java/org/dspace/app/opensearch/OpenSearchControllerTest.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.opensearch;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.xpath;
+
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+
+import org.junit.Test;
+
+/**
+ * Integration test to test the /opensearch endpoint
+ * (Class has to start or end with IT to be picked up by the failsafe plugin)
+ *
+ * @author Oliver Goldschmidt (o dot goldschmidt at tuhh dot de)
+ */
+public class OpenSearchControllerTest extends AbstractControllerIntegrationTest {
+
+    @Test
+    public void searchAtomTest() throws Exception {
+        //When we call the root endpoint
+        getClient().perform(get("/opensearch/search")
+                                .param("query", "cats"))
+                   //The status has to be 200 OK
+                   .andExpect(status().isOk())
+                   //We expect the content type to be "application/atom+xml;charset=UTF-8"
+                   .andExpect(content().contentType("application/atom+xml;charset=UTF-8"))
+        ;
+    }
+
+    /* HTML is an open issue in OpenSearch, so skip this test at the moment
+    @Test
+    public void searchHtmlTest() throws Exception {
```

Comment:
you can flag the test as Ignored instead than comment it out. This will be noted by statistics tools

Reply:
done